corehq/ex-submodules/dimagi/utils/post.py

```diff
@@ -1,28 +1,31 @@
-import requests
-import six
+from requests.structures import CaseInsensitiveDict
 
+from corehq.motech.requests import Requests
 
-def simple_post(data, url, content_type="text/xml", timeout=60, headers=None, auth=None, verify=None):
+
+def simple_post(domain, url, data, *, headers, auth, verify,
+                timeout=60, notify_addresses=None, payload_id=None):
     """
     POST with a cleaner API, and return the actual HTTPResponse object, so
     that error codes can be interpreted.
     """
-    if isinstance(data, six.text_type):
-        data = data.encode('utf-8')  # can't pass unicode to http request posts
-    default_headers = requests.structures.CaseInsensitiveDict({
-        "content-type": content_type,
+    default_headers = CaseInsensitiveDict({
+        "content-type": "text/xml",
         "content-length": str(len(data)),
     })
-    if headers:
-        default_headers.update(headers)
+    default_headers.update(headers)
     kwargs = {
         "headers": default_headers,
         "timeout": timeout,
     }
-    if auth:
-        kwargs["auth"] = auth
-
-    if verify is not None:
-        kwargs["verify"] = verify
-
-    return requests.post(url, data, **kwargs)
+    requests = Requests(
+        domain,
+        base_url='',
+        username=None,
+        password=None,
+        verify=verify,
+        notify_addresses=notify_addresses,
+        payload_id=payload_id,
+    )
+    # Use ``send_request()`` instead of ``post()`` to pass ``auth``.
```

Comment:
I'm like 98% sure `.post()` accepts an `auth` parameter - `get` and `put` definitely do.

Reply:
You are right, and I should have been clearer. `requests.post()` does accept an `auth` parameter. But `corehq.motech.Requests.post()` [assigns `self.auth` to that parameter here](https://github.com/dimagi/commcare-hq/blob/6646bda0727e8bc2f8b6eb6ccc5725f7d034e289/corehq/motech/requests.py#L166).

:thinking: An alternative could be ...
```python
    requests.auth = auth
    return requests.post(url, data=data, **kwargs)
```
... although this will need to change regardless, when [Requests starts using AuthManager](https://github.com/dimagi/commcare-hq/pull/27531/commits/fd612d48ad8c02476374735e5e3077259a965e68). :confounded: Doh.